src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs

```diff
@@ -271,11 +272,12 @@ public string Path
                 value = (value == null) ? string.Empty : value;
                 if (!string.Equals(_directory, value, PathInternal.StringComparison))
                 {
-                    if (!Directory.Exists(value))
-                    {
-                        throw new ArgumentException(SR.Format(SR.InvalidDirName, value));
-                    }
+                    if (value.Length == 0)
+                        throw new ArgumentException(SR.Format(SR.InvalidDirName, value), nameof(this.Path));
```

Comment:
Why does this check need to be separated out?

Reply:
So he can use a different message.

Reply:
I don't understand why a different message is needed. Why is empty string more special as an invalid name than others that Exists returns false for, in particular for ones that are similarly invalid like those that contain invalid for system characters? I don't see this change actually improving understandability. What an I missing?